src/ott/neural/models/base_models.py

```diff
@@ -0,0 +1,60 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+from typing import Optional
+
+import flax.linen as nn
+import jax.numpy as jnp
+
+__all__ = ["BaseNeuralVectorField", "BaseRescalingNet"]
+
+
+class BaseNeuralVectorField(nn.Module, abc.ABC):
+  """Base class for neural vector field models."""
```

Comment:
some maths? e.g. :math:`v: \mathbb{R}\times \mathbb{R}^d\rightarrow \mathbb{R}^d`, and maybe describe possible use within and ODE